pkg/apis/core/validation/validation.go

```diff
@@ -4338,6 +4338,12 @@ func ValidateService(service *core.Service, allowAppProtocol bool) field.ErrorLi
 		}
 	}
 
+	if service.Spec.AllocateLoadBalancerNodePorts != nil {
```

Comment:
I think we can make this into a single line and it's probably slightly more efficient to do the Type check first:
```
if service.Spec.Type != core.ServiceTypeLoadBalancer &&  service.Spec.AllocateLoadBalancerNodePorts != nil  {
     allErrs = append(allErrs, field.Forbidden(specPath.Child("allocateLoadBalancerNodePorts"), "may only be used when `type` is 'LoadBalancer'")) 
}
```

Reply:
Actually, I don't think order really matters

Reply:
I'll make it one-line but keep the order. I think like 98% of K8s cluster will never set AllocateLoadBalancerNodePorts and then it make sense to check that first. For the remaining 2% of us it's a great feature though :smile:

Reply:
Unless of course GCE/GKE will use it since they don't load-balance to nodeports (or so I've heard).